impl/src/main/java/org/ehcache/internal/store/tiering/CacheStore.java

```diff
@@ -383,13 +383,15 @@ public void disableStoreEventNotifications() {
       }
     }
 
+    @SuppressWarnings("unchecked")
     @Override
-    public void releaseStore(Store<?, ?> resource) {
+    public void releaseStore(Store<?, ?> resource) throws CacheAccessException {
       Map.Entry<CachingTier.Provider, AuthoritativeTier.Provider> entry = providersMap.get(resource);
       if (entry == null) {
         throw new IllegalArgumentException("Given store is not managed by this provider : " + resource);
       }
       CacheStore cacheStore = (CacheStore) resource;
+      cacheStore.flushAll((Store)cacheStore.realCachingTier);
```

Comment:
Had a chat with Rishabh on this, I believe the right path is to make use of `CachingTier.invalidate()`